src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -175,30 +175,40 @@ def update_ag_frontend_port(instance, parent, item_name, port=None): # pylint: d
     return parent
 
 def create_ag_http_listener(resource_group_name, application_gateway_name, item_name,
-                            frontend_ip, frontend_port, ssl_cert=None, no_wait=False):
+                            frontend_ip, frontend_port, host_name=None, ssl_cert=None,
+                            no_wait=False):
     from azure.mgmt.network.models import ApplicationGatewayHttpListener
     ncf = _network_client_factory()
     ag = ncf.application_gateways.get(resource_group_name, application_gateway_name)
     new_listener = ApplicationGatewayHttpListener(
         name=item_name,
         frontend_ip_configuration=SubResource(frontend_ip),
         frontend_port=SubResource(frontend_port),
+        host_name=host_name,
+        require_server_name_indication=True if ssl_cert and host_name else None,
```

Comment:
I used else None because that's the default for the constructor so in case specifying "False" for the N/A situation would cause problems, I just stuck to True or None.  You can't have an SSL based endpoint with it set to False anyways (at least not one that would work).